aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogView.java

```diff
@@ -0,0 +1,883 @@
+      catalog().dropTable(table);
+    }
+  }
+
+  private void deleteTableIgnoringNotFound(String tableName) {
+    try {
+      GLUE.deleteTable(
+          DeleteTableRequest.builder()
+              .catalogId(awsProperties().glueCatalogId())
+              .databaseName("ns")
+              .name(tableName)
+              .build());
+    } catch (EntityNotFoundException e) {
+      // Entity doesn't exist - this is one of the expected states
+    }
+  }
+
+  @AfterAll
+  public static void afterClass() {
+    AwsIntegTestUtil.cleanGlueCatalog(GLUE, NAMESPACES);
+    AwsIntegTestUtil.cleanS3GeneralPurposeBucket(S3, TEST_BUCKET_NAME, TEST_PATH_PREFIX);
+  }
+
+  @Override
+  protected GlueCatalog catalog() {
+    return glueCatalog;
+  }
+
+  @Override
+  protected Catalog tableCatalog() {
+    return glueCatalog;
+  }
+
+  protected static String getRandomName() {
+    return UUID.randomUUID().toString().replace("-", "");
+  }
+
+  protected String createNamespace() {
+    String namespace = getRandomName();
+    NAMESPACES.add(namespace);
+    glueCatalog.createNamespace(Namespace.of(namespace));
+    return namespace;
+  }
+
+  protected AwsProperties awsProperties() {
+    return new AwsProperties();
+  }
+
+  @Override
+  @Test
+  public void completeCreateView() {
+    TableIdentifier identifier = TableIdentifier.of("ns", "view");
+
+    if (requiresNamespaceCreate()) {
+      catalog().createNamespace(identifier.namespace());
+    }
+
+    assertThat(catalog().viewExists(identifier)).as("View should not exist").isFalse();
+
+    String location = String.format("s3://%s/%s/ns/view", TEST_BUCKET_NAME, TEST_PATH_PREFIX);
+
+    View view =
+        catalog()
+            .buildView(identifier)
+            .withSchema(SCHEMA)
+            .withDefaultNamespace(identifier.namespace())
+            .withDefaultCatalog(catalog().name())
+            .withQuery("spark", "select * from ns.tbl")
+            .withQuery("trino", "select * from ns.tbl using X")
+            .withProperty("prop1", "val1")
+            .withProperty("prop2", "val2")
+            .withLocation(location)
+            .create();
+
+    assertThat(view).isNotNull();
+    assertThat(catalog().viewExists(identifier)).as("View should exist").isTrue();
+    assertThat(((BaseView) view).operations().current().metadataFileLocation()).isNotNull();
+
+    if (!overridesRequestedLocation()) {
+      assertThat(view.location()).isEqualTo(location);
+    } else {
+      assertThat(view.location()).isNotNull();
+    }
+
+    // validate view settings
+    assertThat(view.uuid())
+        .isEqualTo(UUID.fromString(((BaseView) view).operations().current().uuid()));
+    assertThat(view.name()).isEqualTo(ViewUtil.fullViewName(catalog().name(), identifier));
+    assertThat(view.properties()).containsEntry("prop1", "val1").containsEntry("prop2", "val2");
+    assertThat(view.history())
+        .hasSize(1)
+        .first()
```

Comment:
Nit: use `singleElement()` rather than `hasSize(1).first()`
```suggestion
    assertThat(view.history())
        singleElement()
```

Reply:
addressed